include/aspect/material_model/interface.h

```diff
@@ -516,6 +516,39 @@ namespace aspect
     };
 
 
+    /**
+     * A class for additional output fields to be added to the RHS of the
+     * Stokes system, which can be attached to the
+     * MaterialModel::MaterialModelOutputs structure and filled in the
+     * MaterialModel::Interface::evaluate() function.
+     */
+    template<int dim>
+    class AdditionalMaterialOutputsStokesRHS: public AdditionalMaterialOutputs<dim>
+    {
+      public:
+        AdditionalMaterialOutputsStokesRHS(const unsigned int n_points,
+                                           const unsigned int n_stokes_components)
+          : rhs (n_points, std::vector<double>(n_stokes_components))
+        {}
+
+        virtual ~AdditionalMaterialOutputsStokesRHS()
+        {}
+
+        virtual void average (const MaterialAveraging::AveragingOperation /*operation*/,
+                              const FullMatrix<double>  &/*projection_matrix*/,
+                              const FullMatrix<double>  &/*expansion_matrix*/)
+        {
+          // TODO: not implemented
+        }
+
+        /**
+         * RHS (force) for the Stokes system with the following components:
+         * dim components for velocity, 1 component for pressure, 1 component
+         * for compaction pressure (if melt transport is enabled).
+         */
+        std::vector<std::vector<double> > rhs;
```

Comment:
You call this class `StokesRHS`, so I suspect that the force terms should be `Tensor<1,dim>` objects, assuming you're talking about the force balance equation. If you want additional terms for the mass balance equation, maybe have another `std::vector<double>`.

If you really wanted to be generic and add terms for other equations as well, I would suggest to not document the order here, but use the same ordering as used by the introspection object (to which both the material models and assemblers have access).